centreon/src/CentreonRemote/Domain/Service/ConfigurationWizard/ServerConnectionConfigurationService.php

```diff
@@ -277,4 +276,72 @@ protected function isRemote(): bool
     {
         return false;
     }
+
+    /**
+     * @param array<string, array<int, string[]>> $brokerInfos
+     *
+     * @throws \Throwable
+     *
+     * @return array<string, array<int, string[]>>
+     */
+    protected function saveCredentialInVault(array $brokerInfos): array
+    {
+        $kernel = Kernel::createForWeb();
+        /** @var ReadVaultConfigurationRepositoryInterface $readVaultConfigurationRepository */
+        $readVaultConfigurationRepository = $kernel->getContainer()->get(
+            ReadVaultConfigurationRepositoryInterface::class
+        );
+        /** @var FeatureFlags $featureFlagManager */
+        $featureFlagManager = $kernel->getContainer()->get(FeatureFlags::class);
+
+        /** @var WriteVaultRepositoryInterface $writeVaultRepository */
+        $writeVaultRepository = $kernel->getContainer()->get(WriteVaultRepositoryInterface::class);
+        $writeVaultRepository->setCustomPath(AbstractVaultRepository::BROKER_VAULT_PATH);
+
+        $vaultConfiguration = $readVaultConfigurationRepository->find();
+
+        if (
+            ! $featureFlagManager->isEnabled('vault')
+            || ! $featureFlagManager->isEnabled('vault_broker')
+            || $vaultConfiguration === null
+        ) {
+            return $brokerInfos;
+        }
+
+        foreach ($brokerInfos as $key => $inputOutput) {
+            $inputOutputName = null;
+            $credentialKey = null;
+            $credentialValue = null;
+            foreach ($inputOutput as $index => $row) {
+                if ($row['config_key'] === 'name') {
+                    $inputOutputName = $row['config_value'];
+                }
+                if ($row['config_key'] === 'db_password') {
+                    $credentialKey = $index;
+                    $credentialValue = $row['config_value'];
+                }
+            }
+
+            if (
+                $inputOutputName === null
+                || $credentialKey === null
+                || $credentialValue === null
+            ) {
+                continue;
+            }
+
+            $paths = $writeVaultRepository->upsert(
+                $this->uuid,
+                ["{$inputOutputName}_db_password" => $credentialValue]
+            );
+
+            if (end($paths) !== false) {
+                $this->uuid = $this->getUuidFromPath(end($paths));
```

Comment:
You call end() method two times, you can create a variable to store the return of end()  to optimize the code

Reply:
ok